unstructured/ingest/connector/fsspec.py

```diff
@@ -111,13 +124,77 @@ def get_file(self):
             **self.config.get_access_kwargs(),
         )
         logger.debug(f"Fetching {self} - PID: {os.getpid()}")
-        fs.get(rpath=self.remote_file_path, lpath=self._tmp_download_file().as_posix())
+        try:
+            fs.get(rpath=self.remote_file_path, lpath=self._tmp_download_file().as_posix())
+            self.file_exists = True
+        except Exception as e:
+            if isinstance(e, FileNotFoundError):
+                self.file_exists = False
+            raise
+        self.get_file_metadata()
+
+    def _catch_not_implemented(self, method) -> Optional[str]:
+        try:
+            value: datetime = method(self.remote_file_path)
+        except NotImplementedError:
+            return None
+        return value.isoformat()
+
+    @requires_dependencies(["fsspec"])
+    def get_file_metadata(self):
+        """Fetches file metadata from the current filesystem."""
+        from fsspec import AbstractFileSystem, get_filesystem_class
+
+        fs: AbstractFileSystem = get_filesystem_class(self.config.protocol)(
+            **self.config.get_access_kwargs(),
+        )
+        self.file_exists = fs.exists(self.remote_file_path)
+        date_created = self._catch_not_implemented(fs.created)
+        date_modified = self._catch_not_implemented(fs.modified)
+        version = str(fs.checksum(self.remote_file_path))
+
+        self.file_metadata = FsspecFileMeta(
+            date_created,
+            date_modified,
+            version,
+        )
```

Comment:
instead of this pattern where we check if a thing exists and otherwise call get_file_metadata, what about instead just make `file_metadata` [a cached property](https://docs.python.org/3/library/functools.html#functools.cached_property)?